integration/simulation/simulation_in_mem_test.go

```diff
@@ -24,11 +36,78 @@ func TestInMemoryMonteCarloSimulation(t *testing.T) {
 		SimulationTime:            15 * time.Second,
 		L1EfficiencyThreshold:     0.2,
 		L2EfficiencyThreshold:     0.32,
-		L2ToL1EfficiencyThreshold: 0.34,
+		L2ToL1EfficiencyThreshold: 0.5,
 	}
 
 	params.AvgNetworkLatency = params.AvgBlockDuration / 15
 	params.AvgGossipPeriod = params.AvgBlockDuration / 3
 
 	testSimulation(t, network.NewBasicNetworkOfInMemoryNodes(), params)
 }
+
+// TestMemObscuroRealEthMonteCarloSimulation runs the simulation against a ganache network
+// 1. Install ganache -> npm install ganache --global
+// 2. Run ganache -> rm -rf ganachedb &&  ganache --database.dbPath="./ganachedb"  -l 1024000000000 --wallet.accounts="0x5dbbff1b5ff19f1ad6ea656433be35f6846e890b3f3ec6ef2b2e2137a8cab4ae,0x56BC75E2D63100000" --wallet.accounts="0xb728cd9a9f54cede03a82fc189eab4830a612703d48b7ef43ceed2cbad1a06c7,0x56BC75E2D63100000" --wallet.accounts="0x1e1e76d5c0ea1382b6acf76e873977fd223c7fa2a6dc57db2b94e93eb303ba85,0x56BC75E2D63100000" -p 7545 -g 225
+func TestMemObscuroRealEthMonteCarloSimulation(t *testing.T) {
+	//	t.Skip("test under construction")
+	setupTestLog()
+	deployContract(t)
+
+	helpertypes.IsRealEth = true
+	params := params.SimParams{
+		NumberOfNodes:             2,
+		NumberOfWallets:           2,
+		AvgBlockDuration:          10 * time.Millisecond,
+		SimulationTime:            15 * time.Second,
+		L1EfficiencyThreshold:     0.9, // todo review this
+		L2EfficiencyThreshold:     0.9,
+		L2ToL1EfficiencyThreshold: 0.9,
+	}
+
+	params.AvgNetworkLatency = params.AvgBlockDuration / 15
+	params.AvgGossipPeriod = params.AvgBlockDuration / 2
+
+	testSimulation(t, network.NewNetworkInMemoryGeth(), params)
+}
+
+func deployContract(t *testing.T) {
```

Comment:
@otherview Could we make this function idempotent, so that you could run the sim test against the same Ganache network multiple times? Or is that not going to work?

Reply:
Because of the block count and other height validation steps on the simulation deploying multiple contracts will fail validation.
Our simulation is not prepared for "dirty" blockchains. 

We should definetly change it for long running blockchains but I think that would be further down the line ?